ComposeSamples/CompanionPane/src/main/java/com/microsoft/device/display/samples/companionpane/HomePage.kt

```diff
@@ -67,41 +67,36 @@ enum class ScreenState {
 @Composable
 fun SetupUI(windowInfoRep: WindowInfoRepository) {
     var screenState by remember { mutableStateOf(ScreenState.SinglePortrait) }
-
-    val isPortrait = LocalConfiguration.current.orientation == Configuration.ORIENTATION_PORTRAIT
-    screenState = if (isPortrait) {
-        ScreenState.SinglePortrait
-    } else {
-        ScreenState.SingleLandscape
-    }
-
-    val smallestScreenWidthDp = LocalConfiguration.current.smallestScreenWidthDp
-    val isTablet = smallestScreenWidthDp > SMALLEST_TABLET_SCREEN_WIDTH_DP
-    if (isTablet) {
-        screenState = if (isPortrait) {
-            ScreenState.DualLandscape
-        } else {
-            ScreenState.DualPortrait
-        }
-    }
+    var isAppSpanned by remember { mutableStateOf(false) }
+    var isHingeHorizontal by remember { mutableStateOf(false) }
 
     LaunchedEffect(windowInfoRep) {
         windowInfoRep.windowLayoutInfo
             .collect { newLayoutInfo ->
                 val displayFeatures = newLayoutInfo.displayFeatures
-                val isScreenSpanned = displayFeatures.isNotEmpty()
-                if (isScreenSpanned) {
-                    val foldingFeature = displayFeatures.first() as FoldingFeature
-                    val isVertical = foldingFeature.orientation == FoldingFeature.Orientation.VERTICAL
-                    screenState = if (isVertical) {
-                        ScreenState.DualPortrait
-                    } else {
-                        ScreenState.DualLandscape
+                isAppSpanned = displayFeatures.isNotEmpty()
+                if (isAppSpanned) {
+                    val foldingFeature = displayFeatures.first() as? FoldingFeature
+                    foldingFeature?.let {
+                        isHingeHorizontal = it.orientation == FoldingFeature.Orientation.HORIZONTAL
                     }
                 }
             }
     }
 
+    val isPortrait = LocalConfiguration.current.orientation == Configuration.ORIENTATION_PORTRAIT
+    val smallestScreenWidthDp = LocalConfiguration.current.smallestScreenWidthDp
+    val isTablet = smallestScreenWidthDp > SMALLEST_TABLET_SCREEN_WIDTH_DP
+    val isDualScreen = (isAppSpanned || isTablet)
+
+    screenState =
+        if (isDualScreen) {
+            if (isHingeHorizontal) ScreenState.DualLandscape else ScreenState.DualPortrait
```

Comment:
We also need to consider `isHingeHorizontal` for tablet when the device rotates by checking `LocalConfiguration.current.orientation`. Currently, the layout is wrong for the tablet when the device turns portrait.

Reply:
This is the logic we can bring into `TwoPaneLayout` to support different layouts between single pane and dual panes.